src/main/kotlin/com/team4099/robot2023/subsystems/superstructure/Superstructure.kt

```diff
@@ -0,0 +1,990 @@
+                  GamePiece.CUBE ->
+                    Elevator.TunableElevatorHeights.highCubeHeight.get() +
+                      Elevator.TunableElevatorHeights.cubeDropPosition.get()
+                  GamePiece.CONE ->
+                    Elevator.TunableElevatorHeights.highConeHeight.get() +
+                      Elevator.TunableElevatorHeights.coneDropPosition.get()
+                  else -> 0.0.inches
+                }
+              }
+              else -> 0.0.inches
+            }
+          val extension =
+            when (nodeTier) {
+              NodeTier.HYBRID -> Manipulator.TunableManipulatorStates.lowScoreExtension.get()
+              NodeTier.MID -> Manipulator.TunableManipulatorStates.midScoreExtension.get()
+              NodeTier.HIGH -> Manipulator.TunableManipulatorStates.highScoreExtension.get()
+              else -> 0.0.inches
+            }
+          manipulator.currentRequest =
+            Request.ManipulatorRequest.TargetingPosition(
+              extension, ManipulatorConstants.IDLE_VOLTAGE
+            )
+          elevator.currentRequest = Request.ElevatorRequest.TargetingPosition(scoreHeight)
+        }
+
+        // Transition
+        if (currentRequest is SuperstructureRequest.Score &&
+          groundIntake.isAtTargetedPosition &&
+          manipulator.isAtTargetedPosition &&
+          elevator.isAtTargetedPosition
+        ) {
+          nextState =
+            when (usingGamePiece) {
+              GamePiece.CUBE -> SuperstructureStates.SCORE_CUBE
+              GamePiece.CONE -> SuperstructureStates.SCORE_CONE
+              else -> SuperstructureStates.IDLE
+            }
+        } else if (currentRequest is SuperstructureRequest.Idle) {
+          nextState = SuperstructureStates.IDLE
+        }
+      }
+      SuperstructureStates.SCORE_CUBE -> {
+        // Outputs
+        manipulator.currentRequest =
+          Request.ManipulatorRequest.TargetingPosition(
+            manipulator.armPositionTarget,
+            Manipulator.TunableManipulatorStates.cubeOutVoltage.get()
+          )
+
+        // Transition
+        if (manipulator.isAtTargetedPosition &&
+          (Clock.fpgaTime - lastTransitionTime) >=
+          Manipulator.TunableManipulatorStates.spitTime.get()
+        ) {
+          nextState = SuperstructureStates.IDLE
+        }
+      }
+      SuperstructureStates.SCORE_CONE -> {
+        // Outputs
+        val dropToPosition =
+          when (nodeTier) {
+            NodeTier.HYBRID -> Elevator.TunableElevatorHeights.hybridHeight.get()
+            NodeTier.MID -> Elevator.TunableElevatorHeights.midConeHeight.get()
+            NodeTier.HIGH -> Elevator.TunableElevatorHeights.highConeHeight.get()
+            else -> elevator.elevatorPositionTarget
+          }
+        elevator.currentRequest = Request.ElevatorRequest.TargetingPosition(dropToPosition)
+        if (elevator.isAtTargetedPosition) {
+          manipulator.currentRequest =
```

Comment:
manipulator will need to retract here after scoring the cone, otherwise it will hit the lower nodes as you retract